File: src/shared/news-filter.js
const policy=require('../data/news-filter.json');
function relevantNews(item,category){
  try{const url=new URL(item.link);if(policy.blockedArticleIds.includes(url.pathname.split('/').filter(Boolean).pop()))return false;}catch{return false;}
  const strip=value=>String(value||'').replace(/<[^>]*>/g,' ').replace(/\s+/g,' ').trim();
  const title=strip(item.title),description=strip(item.description);
  const byline=new RegExp(category+'\\s*(?:'+policy.bylineRoles.join('|')+')','g');
  const headline=title.replace(byline,'');
  const text=(headline+' '+description.replace(byline,'')).toLowerCase();
  if(category==='씨야')return headline.includes('씨야')||/\bseeya\b/i.test(headline);
  // A name appearing only in a reporter credit or RSS description is insufficient.
  return headline.includes(category)&&policy.artistContext.some(term=>text.includes(term.toLowerCase()));
}
function filterNews(items,category){return (Array.isArray(items)?items:[]).filter(item=>relevantNews(item,category));}
module.exports={filterNews,relevantNews};

File: scripts/news_filter.py
import json
import re
from pathlib import Path
from urllib.parse import urlparse

POLICY = json.loads((Path(__file__).resolve().parents[1] / 'src/data/news-filter.json').read_text(encoding='utf-8'))

def relevant_news(item, category):
    link = urlparse(item.get('link', ''))
    if not link.scheme or not link.netloc:
        return False
    if link.path.rstrip('/').split('/')[-1] in POLICY['blockedArticleIds']:
        return False
    def strip(value):
        return re.sub(r'\s+', ' ', re.sub(r'<[^>]*>', ' ', str(value or ''))).strip()
    byline = re.compile(re.escape(category) + r'\s*(?:' + '|'.join(POLICY['bylineRoles']) + ')')
    title = byline.sub('', strip(item.get('title')))
    description = byline.sub('', strip(item.get('description')))
    if category == '씨야':
        return '씨야' in title or re.search(r'\bseeya\b', title, re.I) is not None
    text = (title + ' ' + description).lower()
    return category in title and any(term.lower() in text for term in POLICY['artistContext'])
